src/presentation/http/router/note.test.ts

```diff
@@ -49,4 +52,80 @@ describe('Note API', () => {
       expect(body).toStrictEqual({ message: 'Note not found' });
     });
   });
-});
+
+  describe('GET note/:notePublicId ', () => {
+    test('Returns note by public id with 200 status when note is publicly available', async () => {
+      const expectedStatus = 200;
+      const correctID = 'Pq1T9vc23Q';
+
+      const expectedNote = {
+        'id': 2,
+        'publicId': 'Pq1T9vc23Q',
+        'creatorId': 1,
+        'content': null,
+        'createdAt': '2023-10-16T13:49:19.000Z',
+        'updatedAt': '2023-10-16T13:49:19.000Z',
+      };
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual(expectedNote);
+    });
+
+    test('Returns 403 when public access is disabled, user is not creator of the note', async () => {
+      const expectedStatus = 403;
+
+      const notPublicNote = notes.find(newNote => {
+        const settings = noteSettings.find(ns => ns.note_id === newNote.id);
+
+        return settings!.is_public === false;
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${notPublicNote!.public_id}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'Permission denied' });
+    });
+
+    test('Returns 406 when the id  does not exist', async () => {
+      const expectedStatus = 406;
+      const nonexistentId = 'ishvm5qH84';
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${nonexistentId}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+    });
+
+    test('Returns 406 when passed id is incorrect', async () => {
+      const expectedStatus = 406;
+      const incorrectCharId = 'PR^B{@,&S!';
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note/${incorrectCharId}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+    });
```

Comment:
seems like this these is not working properly because we have a different  response for incorrect id. Try sending request to an API and you will see it.

Reply:
I tried to find an id that would give a different answer, but unfortunately any query gives either "Note not found" or status 400

Reply:
```bash
curl -X 'GET' 'https://api.notex.so/note/aaabbb' -H 'accept: */*'
```

Reply:
400 is a correct response for this case.